Custom Phase-2 L1 Trigger NanoAOD

artlbv opened the pull request:
#### PR description:

This PR adds the Phase-2 L1 Trigger custom Nano version from a standalone repo https://github.com/cms-l1-dpg/Phase2-L1Nano/ to CMSSW. It allows storing the L1 and P2GT objects and also adds some gen collections used for Phase-2 Object and Menu studies.

This nano flavour is added to `autoNANO` as `@Phase2L1DPG` and also two upgrade workflows are implemented to test/run this nano after the full `DigiTrigger` chain after the complete L1: 
* `.781` - produces NANO in addition to FEVTDEBUG
* `.782` - produces only NANO
I also added the `Phase2L1DPGwithGen` flavour which includes the Gen and some RECO tables used in the MenuTools of the L1 DPG which are normally contained in the MINIAOD.

Thus these wfs can be used for testing Phase-2 L1 PRs in future @aloeliger @slaurila @RobertJWard @jngadiub.

Some more documentation is in the `DPGAnalysis/Phase2L1TNanoAOD/README.md`.

#### PR validation:

Tested with the workflows `29634.781` and `29634.782` and manually with `cmsDriver.py -s NANO:@Phase2L1DPG` on 15X TTbar relvals.

artlbv committed:
Add Phase2 L1Nano as separate DPG Package from https://github.com/cms-l1-dpg/Phase2-L1Nano/

artlbv committed:
Add Phase2 L1 nano into autoNANO

artlbv committed:
Add Phase2 L1 Nano workflows

artlbv committed:
Commen out HPSTaus from Ph2L1nano

artlbv committed:
Fix P2L1nano wfs

artlbv committed:
Moved P2GT results converter to P2L1nano folder

artlbv committed:
Clean up and update readme

artlbv committed:
Some more cleanup

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"06b48dcac0f85277848818e271d6bddde7ea2ad1":{"files":["DPGAnalysis/Phase2L1TNanoAOD/BuildFile.xml","DPGAnalysis/Phase2L1TNanoAOD/README.md","DPGAnalysis/Phase2L1TNanoAOD/plugins/BuildFile.xml","DPGAnalysis/Phase2L1TNanoAOD/plugins/L1TableProducer.cc","DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2GTtables_cff.py","DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nano_cff.py","DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nanotables_cff.py"],"squashed":false,"time":1737663585},"1143c02e8c3e57cbf15dd3895baad9bb5ee73eb0":{"files":["Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py"],"squashed":false,"time":1737663633},"1cb88c3b42abc5af98a144773e6e1ad46634164f":{"files":["PhysicsTools/NanoAOD/python/autoNANO.py"],"squashed":false,"time":1737663608},"1f358fe8d7e42f89c95b045815c3b457bf0518ba":{"files":["DPGAnalysis/Phase2L1TNanoAOD/plugins/P2GTTriggerResultsConverter.cc"],"squashed":false,"time":1737667289},"1f5ca7ce4f6913e89434e02cb96fc3b544cd15aa":{"files":["DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nanotables_cff.py"],"squashed":false,"time":1737665503},"87866bae26bec58e3f82c6b839bc9bb2ca7c3b3a":{"files":["Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py"],"squashed":false,"time":1737666810},"e4be0be9cd2ce1e1a7acf2a0fb421c7dd6021117":{"files":["DPGAnalysis/Phase2L1TNanoAOD/README.md","DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nano_cff.py","DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nanotables_cff.py"],"squashed":false,"time":1737669108},"ec8c2b5bf2c387474a0abb4ae43f09c1133f4693":{"files":["DPGAnalysis/Phase2L1TNanoAOD/python/l1tPh2Nanotables_cff.py","PhysicsTools/NanoAOD/python/autoNANO.py"],"squashed":false,"time":1737709002}},"emoji":{},"last_seen_sha":"ec8c2b5bf2c387474a0abb4ae43f09c1133f4693","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47178/43417

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #47069, #47090, #47113
  - File  PhysicsTools/NanoAOD/python/autoNANO.py  modified in PR(s): #47069, #47102

artlbv commented:
I added the new package `DPGAnalysis/Phase2L1TNanoAOD` to `l1,xpog` in this PR: https://github.com/cms-sw/cms-bot/pull/2420

artlbv commented:
@aloeliger @Moanwar @hqucms @ftorrresd could you please either launch the tests and/or give me test rights?
thanks!

Moanwar commented:
allow @artlbv test rights

artlbv commented:
@cmsbuild please test

artlbv commented:
I forgot to add the new workflows, so here they are:

@cmsbuild test parameters:

 * relvals_opt = --what upgrade
 * workflows = 29634.78, 29634.781, 29634.782

ftorrresd commented:
enable nano

ftorrresd commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d72480/44161/summary.html
**COMMIT**: ec8c2b5bf2c387474a0abb4ae43f09c1133f4693
**CMSSW**: CMSSW_15_0_X_2025-02-03-1100/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/47178/44161/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 2 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-03-1100+d72480/66748/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4016938
 - DQMHistoTests: [Total failures: 97](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-03-1100+d72480/66748/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4016821
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 218 log files, 189 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-03-1100+d72480/66748/triggerResults): no differences found


## NANO Comparison Summary



**Summary**:
 - You potentially [removed 939 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-03-1100+d72480/66746/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-03-1100+d72480/66746/validateJR.html)
 - DQMHistoTests: Total files compared: 21
 - DQMHistoTests: Total histograms compared: 75219
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-03-1100+d72480/66746/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 75219
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 20 files compared)
 - Checked 106 log files, 61 edm output root files, 21 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-03-1100+d72480/66746/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001 | 3.114 | 3.114 | 0.000 ( +0.0% ) | 5.54 | 6.42 | -13.8% | 2.547 | 2.550 |
| 2500.002 | 3.230 | 3.230 | 0.000 ( +0.0% ) | 5.08 | 5.72 | -11.1% | 2.984 | 2.984 |
| 2500.003 | 3.171 | 3.171 | 0.000 ( +0.0% ) | 5.26 | 6.01 | -12.6% | 2.963 | 2.956 |
| 2500.011 | 1.644 | 1.644 | 0.000 ( +0.0% ) | 8.50 | 10.27 | -17.2% | 2.634 | 2.627 |
| 2500.012 | 2.184 | 2.184 | 0.000 ( +0.0% ) | 5.33 | 6.00 | -11.2% | 2.813 | 2.818 |
| 2500.013 | 2.000 | 2.000 | 0.000 ( +0.0% ) | 7.15 | 8.44 | -15.2% | 2.718 | 2.727 |
| 2500.021 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.85 | 2.08 | -11.1% | 2.602 | 2.603 |
| 2500.022 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.77 | 1.99 | -11.2% | 2.598 | 2.604 |
| 2500.023 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.72 | 1.89 | -9.4% | 2.474 | 2.470 |
| 2500.024 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.41 | 1.56 | -9.8% | 2.700 | 2.703 |
| 2500.031 | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.55 | 1.76 | -11.9% | 2.654 | 2.657 |
| 2500.032 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.65 | 1.82 | -9.6% | 2.627 | 2.620 |
| 2500.033 | 0.037 | 0.037 | 0.000 ( +0.0% ) | 1.54 | 1.70 | -9.5% | 2.704 | 2.707 |
| 2500.034 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.53 | 1.71 | -10.3% | 2.688 | 2.682 |
| 2500.101 | 2.847 | 2.847 | 0.000 ( +0.0% ) | 12.83 | 16.54 | -22.4% | 2.636 | 2.642 |
| 2500.111 | 1.465 | 1.465 | 0.000 ( +0.0% ) | 23.88 | 31.89 | -25.1% | 2.334 | 2.326 |
| 2500.112 | 1.885 | 1.885 | 0.000 ( +0.0% ) | 18.71 | 25.59 | -26.9% | 2.405 | 2.408 |
| 2500.131 | 0.750 | 0.750 | 0.000 ( +0.0% ) | 29.41 | 37.66 | -21.9% | 1.515 | 1.508 |
| 2500.201 | 2.676 | 2.676 | 0.000 ( +0.0% ) | 10.16 | 13.90 | -26.9% | 2.210 | 2.208 |
| 2500.211 | 1.833 | 1.833 | 0.000 ( +0.0% ) | 19.96 | 27.53 | -27.5% | 2.402 | 2.400 |
| 2500.212 | 2.229 | 2.229 | 0.000 ( +0.0% ) | 17.69 | 22.50 | -21.4% | 2.494 | 2.495 |
| 2500.221 | 2.038 | 2.038 | 0.000 ( +0.0% ) | 10.63 | 14.61 | -27.3% | 2.116 | 2.120 |
| 2500.222 | 3.482 | 3.482 | 0.000 ( +0.0% ) | 9.43 | 13.25 | -28.8% | 2.220 | 2.220 |
| 2500.223 | 9.493 | 9.493 | 0.000 ( +0.0% ) | 3.45 | 4.30 | -19.7% | 2.295 | 2.290 |
| 2500.224 | 6.547 | 6.547 | 0.000 ( +0.0% ) | 1.16 | 1.42 | -17.9% | 2.238 | 2.288 |
| 2500.225 | 6.594 | 6.594 | 0.000 ( +0.0% ) | 1.10 | 1.31 | -16.3% | 2.447 | 2.505 |
| 2500.226 | 3.175 | 3.175 | 0.000 ( +0.0% ) | 11.06 | 13.76 | -19.6% | 2.219 | 2.212 |
| 2500.227 | 1.442 | 1.442 | 0.000 ( +0.0% ) | 18.91 | 23.99 | -21.2% | 1.440 | 1.442 |
| 2500.228 | 3.959 | 3.959 | 0.000 ( +0.0% ) | 8.03 | 9.24 | -13.0% | 2.316 | 2.321 |
| 2500.231 | 1.457 | 1.457 | 0.000 ( +0.0% ) | 15.36 | 23.00 | -33.2% | 2.292 | 2.290 |
| 2500.232 | 2.489 | 2.489 | 0.000 ( +0.0% ) | 15.44 | 21.47 | -28.1% | 2.390 | 2.399 |
| 2500.233 | 4.988 | 4.988 | 0.000 ( +0.0% ) | 4.14 | 6.31 | -34.4% | 2.472 | 2.468 |
| 2500.234 | 3.884 | 3.884 | 0.000 ( +0.0% ) | 1.57 | 1.80 | -12.7% | 2.188 | 2.434 |
| 2500.235 | 3.916 | 3.916 | 0.000 ( +0.0% ) | 1.48 | 1.69 | -12.5% | 2.372 | 2.634 |
| 2500.236 | 2.278 | 2.278 | 0.000 ( +0.0% ) | 15.00 | 22.72 | -34.0% | 2.391 | 2.394 |
| 2500.237 | 1.018 | 1.018 | 0.000 ( +0.0% ) | 32.44 | 35.60 | -8.9% | 1.402 | 1.460 |
| 2500.238 | 2.466 | 2.466 | 0.000 ( +0.0% ) | 16.22 | 17.51 | -7.4% | 2.248 | 2.487 |
| 2500.241 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 4.84 | 6.43 | -24.7% | 1.921 | 1.927 |
| 2500.242 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 1.51 | 1.58 | -4.5% | 1.728 | 1.727 |
| 2500.243 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 12.75 | 15.82 | -19.4% | 1.064 | 1.061 |
| 2500.244 | 486.016 | 486.016 | 0.000 ( +0.0% ) | 1.10 | 1.15 | -4.3% | 1.689 | 1.685 |
| 2500.245 | 826.413 | 826.413 | 0.000 ( +0.0% ) | 1.48 | 1.55 | -4.6% | 1.668 | 1.695 |
| 2500.251 | 645.314 | 645.314 | 0.000 ( +0.0% ) | 1.63 | 1.68 | -3.0% | 1.780 | 1.788 |
| 2500.901 | 1.819 | 1.819 | 0.000 ( +0.0% ) | 43.68 | 45.66 | -4.3% | 1.444 | 1.447 |
| 2500.902 | 1.665 | 1.665 | 0.000 ( +0.0% ) | 45.31 | 49.19 | -7.9% | 1.337 | 1.336 |
| 2500.911 | 14.345 | 14.345 | 0.000 ( +0.0% ) | 8.56 | 8.39 | +2.0% | 1.086 | 1.087 |
| 2500.912 | 0.199 | 0.240 | -0.041 ( -16.9% ) | 3.41 | 3.55 | -3.9% | 0.846 | 0.844 |
| 2500.913 | 0.110 | 0.110 | 0.000 ( +0.0% ) | 2.67 | 2.63 | +1.4% | 0.852 | 0.851 |

artlbv commented:
@cmsbuild test parameters:

relvals_opt = --what upgrade
workflows = 29634.78, 29634.781, 29634.782

aloeliger commented:
test parameters:
 - workflows = 29634.78, 29634.781, 29634.782
 - relvals_opt = --what cleanedupgrade,standard,highstats,pileup,generator,extendedgen,production,identity,ged,machine,premix,nano,gpu,2017,Run4

aloeliger commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d72480/44225/summary.html
**COMMIT**: ec8c2b5bf2c387474a0abb4ae43f09c1133f4693
**CMSSW**: CMSSW_15_0_X_2025-02-05-1100/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/47178/44225/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [29634.782](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d72480/44225/runTheMatrix-results/29634.782_TTbar_14TeV+Run4D110_L1CompleteOnlyNano)
```
----- Begin Fatal Exception 05-Feb-2025 17:52:48 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'NANOAODSIMoutput_step'
   [2] Prefetching for module NanoAODOutputModule/'NANOAODSIMoutput'
   [3] Calling method for module NPUTablesProducer/'puTable'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Vertex>
Looking for module label: offlineSlimmedPrimaryVertices
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "TryToContinue = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
- [29634.781](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d72480/44225/runTheMatrix-results/29634.781_TTbar_14TeV+Run4D110_L1CompleteWithNano)
```
----- Begin Fatal Exception 05-Feb-2025 17:52:50 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'NANOAODSIMoutput_step'
   [2] Prefetching for module NanoAODOutputModule/'NANOAODSIMoutput'
   [3] Calling method for module NPUTablesProducer/'puTable'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Vertex>
Looking for module label: offlineSlimmedPrimaryVertices
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "TryToContinue = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```


## NANO Comparison Summary



**Summary**:
 - You potentially [added 198 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-05-1100+d72480/66793/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-05-1100+d72480/66793/validateJR.html)
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 89591
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-05-1100+d72480/66793/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 89591
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 22 files compared)
 - Checked 113 log files, 65 edm output root files, 23 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-02-05-1100+d72480/66793/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001 | 3.110 | 3.110 | 0.000 ( +0.0% ) | 5.78 | 6.03 | -4.2% | 2.561 | 2.571 |
| 2500.002 | 3.227 | 3.227 | 0.000 ( +0.0% ) | 5.16 | 5.46 | -5.3% | 2.994 | 3.008 |
| 2500.003 | 3.167 | 3.167 | 0.000 ( +0.0% ) | 5.43 | 5.64 | -3.8% | 2.977 | 2.987 |
| 2500.011 | 1.642 | 1.642 | 0.000 ( +0.0% ) | 9.44 | 9.77 | -3.4% | 2.653 | 2.651 |
| 2500.012 | 2.180 | 2.180 | 0.000 ( +0.0% ) | 5.49 | 5.73 | -4.2% | 2.835 | 2.850 |
| 2500.013 | 1.997 | 1.997 | 0.000 ( +0.0% ) | 7.60 | 7.93 | -4.2% | 2.648 | 2.753 |
| 2500.021 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.56 | 1.78 | -12.4% | 2.502 | 2.677 |
| 2500.022 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.55 | 1.72 | -9.7% | 2.502 | 2.680 |
| 2500.023 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.50 | 1.63 | -8.0% | 2.492 | 2.541 |
| 2500.024 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.16 | 1.31 | -11.1% | 2.630 | 2.784 |
| 2500.031 | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.39 | 1.54 | -9.7% | 2.594 | 2.737 |
| 2500.032 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.44 | 1.55 | -6.8% | 2.551 | 2.694 |
| 2500.033 | 0.037 | 0.037 | 0.000 ( +0.0% ) | 1.34 | 1.49 | -10.1% | 2.618 | 2.772 |
| 2500.034 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.32 | 1.51 | -12.5% | 2.613 | 2.764 |
| 2500.101 | 2.850 | 2.850 | 0.000 ( +0.0% ) | 12.73 | 13.34 | -4.5% | 2.592 | 2.645 |
| 2500.111 | 1.469 | 1.469 | 0.000 ( +0.0% ) | 23.58 | 25.12 | -6.2% | 2.323 | 2.340 |
| 2500.112 | 1.889 | 1.889 | 0.000 ( +0.0% ) | 19.31 | 19.76 | -2.3% | 2.252 | 2.406 |
| 2500.131 | 0.750 | 0.750 | 0.000 ( +0.0% ) | 31.05 | 31.29 | -0.8% | 1.473 | 1.496 |
| 2500.201 | 2.679 | 2.679 | 0.000 ( +0.0% ) | 10.21 | 10.64 | -4.0% | 2.060 | 2.209 |
| 2500.211 | 1.836 | 1.836 | 0.000 ( +0.0% ) | 20.79 | 21.42 | -3.0% | 2.232 | 2.397 |
| 2500.212 | 2.233 | 2.233 | 0.000 ( +0.0% ) | 15.50 | 15.87 | -2.3% | 2.288 | 2.489 |
| 2500.221 | 2.115 | 2.115 | 0.000 ( +0.0% ) | 9.14 | 11.33 | -19.3% | 1.962 | 2.124 |
| 2500.222 | 3.487 | 3.487 | 0.000 ( +0.0% ) | 10.86 | 9.24 | +17.5% | 2.015 | 2.220 |
| 2500.223 | 9.490 | 9.490 | 0.000 ( +0.0% ) | 3.78 | 3.65 | +3.5% | 2.059 | 2.338 |
| 2500.224 | 6.604 | 6.604 | 0.000 ( +0.0% ) | 1.03 | 1.07 | -4.1% | 2.103 | 2.295 |
| 2500.225 | 6.652 | 6.652 | 0.000 ( +0.0% ) | 0.96 | 1.02 | -5.4% | 2.340 | 2.498 |
| 2500.226 | 3.180 | 3.180 | 0.000 ( +0.0% ) | 10.45 | 10.93 | -4.4% | 1.907 | 2.214 |
| 2500.227 | 1.442 | 1.442 | 0.000 ( +0.0% ) | 19.66 | 19.69 | -0.1% | 1.240 | 1.446 |
| 2500.228 | 3.996 | 3.996 | 0.000 ( +0.0% ) | 7.39 | 8.06 | -8.3% | 2.309 | 2.309 |
| 2500.231 | 1.512 | 1.512 | 0.000 ( +0.0% ) | 18.10 | 19.76 | -8.4% | 2.175 | 2.316 |
| 2500.232 | 2.493 | 2.493 | 0.000 ( +0.0% ) | 15.63 | 17.12 | -8.7% | 2.162 | 2.397 |
| 2500.233 | 4.986 | 4.986 | 0.000 ( +0.0% ) | 5.76 | 6.06 | -4.9% | 2.275 | 2.519 |
| 2500.234 | 3.919 | 3.919 | 0.000 ( +0.0% ) | 1.30 | 1.39 | -6.6% | 2.430 | 2.236 |
| 2500.235 | 3.950 | 3.950 | 0.000 ( +0.0% ) | 1.21 | 1.35 | -10.0% | 2.435 | 2.419 |
| 2500.236 | 2.282 | 2.282 | 0.000 ( +0.0% ) | 16.66 | 17.81 | -6.4% | 2.194 | 2.106 |
| 2500.237 | 1.018 | 1.018 | 0.000 ( +0.0% ) | 29.02 | 30.51 | -4.9% | 1.389 | 1.400 |
| 2500.238 | 2.469 | 2.469 | 0.000 ( +0.0% ) | 14.53 | 15.44 | -5.9% | 2.367 | 2.473 |
| 2500.241 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 4.70 | 6.09 | -22.7% | 1.827 | 1.921 |
| 2500.242 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 1.19 | 1.42 | -16.1% | 1.663 | 1.724 |
| 2500.243 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 12.22 | 12.53 | -2.4% | 1.060 | 1.059 |
| 2500.244 | 486.016 | 486.016 | 0.000 ( +0.0% ) | 0.99 | 1.02 | -3.5% | 1.632 | 1.698 |
| 2500.245 | 826.413 | 826.413 | 0.000 ( +0.0% ) | 1.30 | 1.35 | -3.6% | 1.623 | 1.678 |
| 2500.251 | 645.314 | 645.314 | 0.000 ( +0.0% ) | 1.33 | 1.54 | -13.9% | 1.781 | 1.494 |
| 2500.301 | 0.021 | 0.021 | 0.000 ( +0.0% ) | 1.30 | 1.72 | -24.5% | 2.792 | 2.789 |
| 2500.311 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.43 | 1.58 | -9.9% | 2.737 | 2.741 |
| 2500.901 | 1.819 | 1.819 | 0.000 ( +0.0% ) | 32.80 | 43.65 | -24.9% | 1.443 | 1.448 |
| 2500.902 | 1.665 | 1.665 | 0.000 ( +0.0% ) | 39.42 | 44.70 | -11.8% | 1.338 | 1.338 |
| 2500.911 | 14.345 | 14.345 | 0.000 ( +0.0% ) | 6.30 | 8.39 | -24.9% | 1.088 | 1.085 |
| 2500.912 | 0.240 | 0.240 | 0.000 ( +0.0% ) | 2.86 | 2.58 | +10.8% | 0.847 | 0.845 |
| 2500.913 | 0.110 | 0.110 | 0.000 ( +0.0% ) | 1.64 | 2.48 | -34.0% | 0.850 | 0.851 |

artlbv committed:
Fix default L1Nano sequence to not use RECO and Gen objects, update readme

artlbv committed:
Add new Phase2L1Nano wfs to PyRelVal README

artlbv commented:
I modified the `Phase2L1DPG` to not include the Gen and Reco (PV) tables, as the Reco is missing from the workflows I tested here. Instead I added `Phase2L1DPGwithGen` which includes these tables used in the MenuTools of the L1 DPG.

I also added the new wfs .781 and .782 to the PyRelVal readme.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47178/43589

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/README.md  modified in PR(s): #47069, #47113
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #47069, #47090, #47113
  - File  PhysicsTools/NanoAOD/python/autoNANO.py  modified in PR(s): #47069

cmsbuild commented:
Pull request #47178 was updated. @AdrianoDee, @Moanwar, @aloeliger, @cmsbuild, @dickychant, @epalencia, @ftorrresd, @hqucms, @miquork, @srimanob, @subirsarkar can you please check and sign again.

artlbv commented:
@cmsbuild please test

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_15_1_X. Please open a backport if it should also go in to CMSSW_15_0_X.

aloeliger commented:
please abort

aloeliger commented:
please test

aloeliger commented:
+l1

Moanwar commented:
+Upgrade

AdrianoDee commented:
+pdmv

ftorrresd commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @rappoccio, @sextonkennedy, @antoniovilela, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1
